### ISSUE TITLE
Add ability to specify link target

### ISSUE THREAD

**Issue by sdmcraft** [adobe/aem-boilerplate#119]
### Expected Behaviour
Page authors should have the ability to specify the [link target](https://www.w3schools.com/tags/att_a_target.asp) i.e. whether it opens in the same tab/ new tab etc in the authoring document.
 
### Actual Behaviour
Currently there's no such ability.  Developers need to implement custom logic for achieving this. For e.g. https://github.com/hlxsites/mmm-af/commit/9dba40001d9a9c461d92c4a1e13b6c2cc7b5e1b9

### Reproduce Scenario (including but not limited to)
Not Applicable

#### Steps to Reproduce
Not Applicable

#### Platform and Version
All

#### Sample Code that illustrates the problem
Not Applicable

#### Logs taken while reproducing problem
Not Applicable

**Comment by trieloff** [comment]
I don't think this is nearly as common enough to be left to authors. In most cases, sites can formulate rules such as "all external links should open in a new window except for these hostnames", so that no one-off decisions by authors are needed.

**Comment by sdmcraft** [comment]
@trieloff , I logged this ticket based on a [discussion here](https://cq-dev.slack.com/archives/C02DL4SSXCN/p1659166222212469) wherein @davidnuescheler mentioned that this has come more than once and could be considered for inclusion in the bolierplate. If we feel that isn's the case or we need to reconsider, I can close this. Else perhaps an [implementation like this](https://github.com/hlxsites/mmm-af/commit/9dba40001d9a9c461d92c4a1e13b6c2cc7b5e1b9) can be ported here,

**Comment by ramboz** [comment]
So the question seems to be if we can find valid use-cases for internal links to open in new tabs consistently.

From the top of my head, I can think of 2 use cases that could benefit from this:
1. Links to pdf, doc, xls, etc. would benefit from being opened in a new tab all the time as they represent a complete change of context (from an accessibility point-of-view, and would need to be combined with https://www.w3.org/TR/WCAG20-TECHS/G201.html)
2. Links to contextual documentation or glossaries could be opened in a named window/modal with `#` anchors to scroll to the right section immediately. I think that wouldn't work great with the proposed `#_blank` option, as we'd need to combine 2 anchors

And then there is a 3rd… more gray area one:

3. Links to code samples, which may include css/js and would thus be hard to differentiate from the ones you'd want to download forcibly. That's probably more project-specific so harder to define a default behavior. But 2. could also address this.

**Comment by fkakatie** [comment]
Closing as **out of scope for boilerplate**. 

Link target behavior is too opinionated for core boilerplate because:
1. **Localization required** - WCAG-compliant warnings need project-specific text
2. **Behavior varies by project** - Different sites have different rules for when/how links should open
3. **Accessibility concerns** - Forcing `target="_blank"` conflicts with [WCAG 2.1/2.2](https://www.w3.org/TR/WCAG20-TECHS/G201.html) predictability principles

Per boilerplate's philosophy: **disclosure > forced behavior**. Projects should implement link decoration with appropriate visual indicators, accessible text, and behavioral choices for their audience. See [detailed discussion in PR #584](https://github.com/adobe/aem-boilerplate/pull/584).
